src/providers/WorkflowCore.LockProviders.MySQL/MySqlLockProvider.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using WorkflowCore.Interface;
+using System.Data;
+using System.Collections.Generic;
+using System.Threading;
+using MySql.Data.MySqlClient;
+
+namespace WorkflowCore.LockProviders.MySQL
+{
+    public class MySqlLockProvider : IDistributedLockProvider
+    {
+        private const string Prefix = "wfc";
+
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+        private readonly Dictionary<string, MySqlConnection> _locks = new Dictionary<string, MySqlConnection>();
+        private readonly AutoResetEvent _mutex = new AutoResetEvent(true);
+
+        public MySqlLockProvider(string connectionString, ILoggerFactory logFactory)
+        {
+            _logger = logFactory.CreateLogger<MySqlLockProvider>();
+            var csb = new MySqlConnectionStringBuilder(connectionString);
+            csb.Pooling = true;
+            _connectionString = csb.ToString();
+        }
+
+
+        public async Task<bool> AcquireLock(string Id, CancellationToken cancellationToken)
+        {
+            if (_mutex.WaitOne())
+            {
+                try
+                {
+                    var connection = new MySqlConnection(_connectionString);
+                    await connection.OpenAsync(cancellationToken);
+                    try
+                    {
+                        var cmd = connection.CreateCommand();
+                        cmd.CommandText = $"SELECT GET_LOCK('{Prefix}:{Id}', 10)";                                                
```

Comment:
Is that a lease time (10)?

Reply:
10 is the timeout in seconds. It will wait for a maximum of 10 seconds for the query to give a result. If the timeout has exceeded, the query will return -1, which is handled as an exception below.